packages/ts/react-crud/src/data-provider.ts

```diff
@@ -121,6 +125,74 @@ export abstract class DataProvider<TItem> {
   protected abstract fetchFilteredCount(page: DataPage<TItem>): Promise<number | undefined> | number | undefined;
 }
 
+export abstract class AbstractComboBoxDataProvider<TItem> {
+  protected readonly list: ComboBoxFetchCallback<TItem>;
+  protected readonly loadTotalCount?: boolean;
+
+  protected sort: Sort | undefined;
+  protected totalCount: number | undefined;
+  protected filteredCount: number | undefined;
+
+  constructor(list: ComboBoxFetchCallback<TItem>, sort: Sort | undefined) {
+    this.list = list;
+    this.load = this.load.bind(this);
+    this.sort = sort;
+  }
+
+  reset(): void {
+    this.totalCount = undefined;
+    this.filteredCount = undefined;
+  }
+
+  async load(params: ComboBoxDataProviderParams, callback: ComboBoxDataProviderCallback<TItem>): Promise<void> {
+    // Fetch page and filtered count
+    const page = await this.fetchPage(params);
+    this.filteredCount = await this.fetchFilteredCount(page);
+    // Only fetch total count if it's specific in options
+    if (this.loadTotalCount) {
+      this.totalCount = await this.fetchTotalCount(page);
+    }
+
+    // Pass results to the combobox
+    callback(page.items, this.filteredCount);
+  }
```

Comment:
As far as I understand, `ComboBoxDataProvider` type requires that the function is not `async` (doesn't return a `Promise`) since it uses `callback` for that. So, we either should adjust the returned `UseComboBoxDataProviderResult` or the `load` function to avoid unexpected type casting and fulfill TypeScript requirements.

The best approach would be to update the `ComboBoxDataProvider` to accept async functions, for sure, but for now, we can use other ways.

Here I tuned the `load` function to be sync function that calls promise under the hood:
```suggestion
  load(params: ComboBoxDataProviderParams, callback: ComboBoxDataProviderCallback<TItem>): void {
    this.fetchPage(params)
      .then(async (page) => {
        this.filteredCount = await this.fetchFilteredCount(page);
        // Only fetch total count if it's specific in options
        if (this.loadTotalCount) {
          this.totalCount = await this.fetchTotalCount(page);
        }

        // Pass results to the combobox
        callback(page.items, this.filteredCount);
      })
      .catch((error: unknown) => {
        throw error;
      });
  }
```

Reply:
Updated